extensions/applicationinsights-cfgsync-js/src/CfgSyncHelperFuncs.ts

```diff
@@ -39,3 +42,76 @@ export function replaceByNonOverrideCfg<T=IConfiguration & IConfig, T1=NonOverri
     // if errors happen, do nothing
     return cfg;
 }
+
+//                                cdn Mode (cdn Value = B, SDK Config Defaults = C)
+//                   |--------------------------------------------------------------------------|
+//                   |                    | none        | disabled    | enabled     | force     |
+//                   | ------------------ | ----------- | ----------- | ----------- | --------- |
+// | User Mode       | none               | A || C      | B || A || C | B || A || C | B || C    |
+// (user Value = A)  | disabled           | A || C      | B || A || C | A || C      | B || C    |
+//                   | enabled            | A || C      | B || A || C | A || B || C | B || C    |
+//                   | none(blockCdn)     | A || C      | A || C      | A || C      | A || C    |
+//                   | disabled(blockCdn) | A || C      | A || C      | A || C      | A || C    |
+//                   | enabled(blockCdn)  | A || C      | A || C      | A || C      | A || C    |
+export function shouldOptInFeature(field: string, cdnCfg?: ICfgSyncCdnConfig, customOptInDetails?: IFeatureOptIn) {
+    
+    if (!cdnCfg || !cdnCfg.enabled) {
+        return null;
+    }
+ 
+    let featureOptIn= cdnCfg.featureOptIn || {};
+    let cdnFeature = featureOptIn[field] || {mode: CdnFeatureMode.none};
+    let cdnMode = cdnFeature.mode;
+    let cdnFeatureVal = cdnFeature.value;
+    let customOptIn = customOptInDetails || {};
+    let customFeature = customOptIn[field] || {mode: FeatureOptInMode.disable}; // default custom mode is disable
+    let customMode = customFeature.mode;
+    let customFeatureVal = customFeature.cfgValue;
+    let blockCdn = customFeature.blockCdnCfg || false;
+
+    if (blockCdn) {
+        return customFeatureVal;
+    }
+
+    if (cdnMode === CdnFeatureMode.force) {
+        return cdnFeatureVal;
+    }
+
+    if (cdnMode === CdnFeatureMode.disable || (customMode === FeatureOptInMode.none && cdnMode === CdnFeatureMode.enable)) {
```

Comment:
I know we have this is the matrix above, but... based on the excel spreadsheet we created do / should we just return null when the `cdnMode` is `disabled`

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/newylie/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/newylie/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:"Aptos Narrow", sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
.xl65
	{color:white;
	font-weight:700;
	background:black;
	mso-pattern:black none;}
.xl66
	{background:#F2F2F2;
	mso-pattern:black none;}
.xl67
	{color:white;
	font-weight:700;
	background:black;
	mso-pattern:black none;
	white-space:normal;}
-->
</style>
</head>

<body link="#467886" vlink="#96607D">



  |   |   | 1 | 2 | 3 | 4
-- | -- | -- | -- | -- | -- | --
  |   |   |   | CDN |   |  
  |   |   | none | disabled / force off | enabled / allow opt in | force on
  | Not Opted In | none | none | disabled | disabled | enabled
A \|\| C | Opt Out | disabled | disabled | disabled | disabled | enabled
  | Opt In | enabled | enabled | disabled | enabled | enabed
  | blocked | none | none | none | none | none
  |   | disabled | disabled | disabled | disabled | disabled
  |   | enabled | enabled | enabled | enabled | enabled



</body>

</html>